include/grpcpp/security/credentials.h

```diff
@@ -178,6 +178,14 @@ std::shared_ptr<ChannelCredentials> SslCredentials(
 /// services.
 std::shared_ptr<CallCredentials> GoogleComputeEngineCredentials();
 
+/// Builds credentials for use when running in GCE
+///
+/// \warning Only use these credentials when connecting to a Google endpoint.
+/// Using these credentials to connect to any other service may result in this
+/// service being able to impersonate your client for requests to Google
+/// services.
+std::shared_ptr<CallCredentials> GoogleComputeEngineCredentialsWithALTS();
```

Comment:
We don't necessarily need this new API. The only use case in C++ is to let the GoogleDefaultCredentials() above to carry two call credentials.

In Go and Java, the same APIs already allow passing in call credentials because it's needed by the client SDK. The case here is different. So I'm thinking we could make a new GoogleDefaultCredentials() API that creates these two ComputeEngineCredentials inside. It reduces the API surface we have to maintain. WDYT?